components/Layout/index.tsx

```diff
@@ -66,13 +60,28 @@ function LayoutSearch(props: LayoutProps) {
     const modalInvestor = useStore($modalInvestor)
     const loginInfo = useSelector((state: RootState) => state.modal)
     const isLight = useSelector((state: RootState) => state.modal.isLight)
-    const showZilpay = useStore($showZilpay)
 
     const bg = loginInfo.isLight ? 'bglight' : 'bg'
 
     const checkZilpayNetwork = async () => {
         if (loginInfo.zilAddr) {
-            updateShowZilpay(true)
+            const wallet = new ZilPayBase()
+            const zp = await wallet.zilpay()
+            const network = zp.wallet.net
+            if (network !== loginInfo.net) {
```

Comment:
hey we need to verify the ZilPay wallet change even if the network remains the same @devwistkey

Reply:
Hi yes, you're right. I just update and now wallet switch works